STYLE_GUIDE.md

```diff
@@ -0,0 +1,85 @@
+# Style Guide
+
+This style guide provides guidelines for writing and formatting code, commit messages, and branch names to ensure consistency and maintainability across the project.
+
+
+## Table of Contents
+- [Code Archiecture](#code-architecture)
+- [General Guidelines](#general-guidelines)
+- [Commit Message Conventions](#commit-message-conventions)
+- [Branch Naming Conventions](#branch-naming-conventions)
+- [Code Formatting](#code-formatting)
+
+
+## Code Architecture
+
+Djate follows a standard Django project structure with `apps` and `manage.py` in the root directory. Each app contains components such as `views.py`, `models.py`, and `urls.py`, while Celery tasks are defined in `tasks.py` for each application.
+
+The code architecture leverages Django REST Framework (DRF) generics and viewsets for building APIs efficiently, ensuring modularity and efficiency in development.
+
+## General Guidelines
+
+- Write clear, concise, and self-explanatory code.
+- Use meaningful variable and function names.
+- Adhere to the language-specific best practices.
+- Ensure your code is modular and reusable.
+- Avoid code duplication.
+
+## Commit Message Conventions
+
+Commit messages should be clear and descriptive. Follow these conventions for commit messages:
+
+- Use the present tense ("Add feature" not "Added feature").
+- Use the imperative mood ("Fix bug" not "Fixed bug").
+- Keep the subject line (first line) under 50 characters.
+- Separate subject from body with a blank line.
+- Provide a detailed description of the changes in the body.
+- Include references to relevant issues or pull requests.
+
+### Examples:
+
+```
+feat: Add user authentication module
+
+Implemented user login and registration functionalities.
+Updated the database schema to include user credentials.
+```
+
+```
+fix: Resolve crash on app startup
+
+Fixed an issue causing the app to crash on startup due to a null pointer exception.
+Closes #42.
+```
+
+## Branch Naming Conventions
+
+Branch names should be descriptive and follow a consistent pattern. Use the following conventions for naming branches:
+
+- Use lowercase letters and hyphens (`-`) to separate words.
+- Include a prefix to indicate the type of branch:
+  - `feat/` for new features
+  - `fix/` for bug fixes
+  - `chore/` for maintenance tasks
+  - `docs/` for documentation updates
+
+### Examples:
+
+```
+feat/add-user-authentication
+fix/crash-on-startup
+chore/update-dependencies
+docs/improve-readme
+```
+
+## Code Formatting
+
```

Comment:
instead of writing in in paragraph break the context into points and sub-points for clear understanding

correction:
Consistent code formatting improves readability and maintainability. Follow these rules:

Use [black](https://black.readthedocs.io/en/stable/) for automatic code formatting.
   -Default configuration with `max_line_length=79` and double quotes.
  -Target Python 3.11 compatibility.

Use [flake8](https://flake8.pycqa.org/) for linting to ensure code quality.
   -Run `flake8` to catch unused variables and other issues.
   -Check for adherence to PEP 8 and best practices.

Before pushing your code, run the following commansd:
python3 -m black .  # Formats code
python3 -m flake8   # Runs lint checks